languageservice/src/validate.ts

```diff
@@ -217,3 +221,98 @@ async function validateExpression(
     );
   }
 }
+
+function validateChomp(
+  diagnostics: Diagnostic[],
+  token: BasicExpressionToken,
+  parent: TemplateToken | undefined,
+  key: TemplateToken | undefined,
+  validationDefinition: Definition | undefined
+): void {
+  // Not "clip" or "keep" chomp style?
+  if (token.chompStyle !== "clip" && token.chompStyle !== "keep") {
+    return;
+  }
+
+  // No definition? This can happen when the token is in an invalid position or the workflow has parse errors.
+  if (!validationDefinition) {
+    return;
+  }
+
+  // Step "run"?
+  if (parent?.definition?.key === "run-step" && key?.isScalar && key.toString() === "run") {
+    return;
+  }
+
+  // Block scalar indicator, i.e. | or >
+  const scalarIndicator = token.scalarType === Scalar.BLOCK_LITERAL ? "|" : ">";
+
+  const defKey = validationDefinition.key;
+  const parentDefKey = parent?.definition?.key;
+
+  // Error for boolean fields
+  if (
+    defKey === "job-if" ||
+    defKey === "step-if" ||
+    defKey === "step-continue-on-error" ||
+    (parentDefKey === "job-factory" && key?.isScalar && key.toString() === "continue-on-error")
+  ) {
+    diagnostics.push({
+      message: `Block scalar adds trailing newline which breaks boolean evaluation. Use '${scalarIndicator}-' to strip trailing newlines.`,
+      range: mapRange(token.range),
+      severity: DiagnosticSeverity.Error,
+      code: "expression-block-scalar-chomping",
+      source: "github-actions"
+    });
+  }
+  // Error for number fields
+  else if (
+    defKey === "step-timeout-minutes" ||
+    (parentDefKey === "container-mapping" && key?.isScalar && ["ports", "volumes"].includes(key.toString())) ||
+    (parentDefKey === "job-factory" && key?.isScalar && key.toString() === "timeout-minutes")
+  ) {
+    diagnostics.push({
```

Comment:
Again, for numbers we can likely safely trim trailing newlines on the server for all number cases